src/models/recommendation/NewsSource.ts

```diff
@@ -0,0 +1,37 @@
+export interface NewsSource {
+    title: string;
+    description: string;
+    url: string;
+    tags: string[];
+}
+
+export interface Category {
+    name: string;
+    subCategories: SubCategory[];
+}
+
+export interface SubCategory {
+    name: string;
+    list: NewsSource[];
+}
+
+import data from '../../../news.json';
+
+type CategoryMap = { [name: string]: SubCategoryMap };
+type SubCategoryMap = { [name: string]: NewsSource[] };
+
+export const serializeData = (categoryMap: CategoryMap = data) => {
```

Comment:
I think this is not a really good name, because one may associate it with serialization (https://en.wikipedia.org/wiki/Serialization) when in fact it transforms a `CategoryMap` to a list of Categories. Maybe `categoryMapToList` is better.

Also I think this function can be eliminated altogether if we already store the data in `news.json` in a list of categories, that is in the exact format how we want to use it later.

Reply:
right, it did something different previously, but still readJSON would have been better.